llmfoundry/command_utils/data_prep/split_eval_set.py

```diff
@@ -0,0 +1,181 @@
+# Copyright 2024 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+import contextlib
+import logging
+import os
+import re
+from typing import Optional
+
+import composer.utils as utils
+import numpy as np
+
+log = logging.getLogger(__name__)
+
+LOCAL_PATH = 'tmp-t'
+REMOTE_OBJECT_STORE_FILE_REGEX = re.compile(
+    r'^((s3|oci|gs):\/\/|dbfs:\/Volumes\/)[/a-zA-Z0-9 ()_\-.]+$',
+)
+
+
+def get_dataset_format(data_path_folder: str) -> str:
+    """Determine the format of the dataset from the provided data path.
+
+    Args:
+        data_path_folder (str): Path to the training dataset folder
+
+    Returns:
+        str: The format of the dataset
+    """
+    if data_path_folder == LOCAL_PATH:
+        return 'local_file'
+    if REMOTE_OBJECT_STORE_FILE_REGEX.match(data_path_folder):
+        return 'remote_object_store'
+    return 'unknown'
+
+
+TEMP_DIR = 'tmp-split'
```

Comment:
shouldn't we be using a real temp dir? or is there a reason we need to have this not be a real tempdir?

Reply:
tempdirs typically get cleaned up by the system on some cadence, but even if that isn't in the case, it puts it in a unified place for temporary things that are no longer needed, and allows the user to configured where this file is written by configuring their temporary directory env vars (which we want to do in our services)